tests: test_receive_before_flush_errors failing on my laptop

When running the tests locally on my laptop (macOS) I get an error:
```
FAILED tests/model_test.py::test_receive_before_flush_errors - assert 4 == 2
```
due to this line:
https://github.com/HEPData/hepdata/blob/5da5dfe3376370606e266c01a58c05dfdcac29f9/tests/model_test.py#L169
If I print out `caplog.records` I get:
```
[<LogRecord: hepdata.modules.submission.models, 10, /Users/graemewatt/src/hepdata/hepdata/modules/submission/models.py, 183, "Deleting data resource None">, <LogRecord: hepdata.modules.submission.models, 40, /Users/graemewatt/src/hepdata/hepdata/modules/submission/models.py, 190, "Unable to delete data resource with id None whilst deleting data submission id 1. Error was: Instance '<DataResource at 0x7f8ab93b6fd0>' is not persisted">, <LogRecord: hepdata.modules.submission.models, 10, /Users/graemewatt/src/hepdata/hepdata/modules/submission/models.py, 192, "Deleting reviews for data submission 1">, <LogRecord: hepdata.modules.submission.models, 40, /Users/graemewatt/src/hepdata/hepdata/modules/submission/models.py, 200, "Unable to delete review with id None whilst deleting data submission id 1. Error was: Instance '<DataReview at 0x7f8ab93c40b8>' is not persisted">]
```
i.e. 4 entries in the list rather than the expected 2.  The test passes when run as a GitHub Actions [workflow](https://github.com/HEPData/hepdata/runs/1389297468?check_suite_focus=true).

--- a/setup.py
+++ b/setup.py
@@ -34,7 +34,7 @@ readme = open('README.rst').read()
 history = open('CHANGES.rst').read()
 
 tests_require = [
-    'pytest>=5.4.2',
+    'pytest>=6.0.2',
     'pytest-cov>=2.9.0',
     'pytest-flask>=1.0.0',
     'pytest-mock>=3.1.0',

